File: ios/Widgetkit.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(Widgetkit, NSObject)

RCT_EXTERN_METHOD(reloadAllTimelines)
RCT_EXTERN_METHOD(reloadTimelines:(NSString *)ofKind)
RCT_EXTERN_METHOD(getItem:(NSString *)key withAppGroup:(NSString *)appGroup withResolver:(RCTPromiseResolveBlock)resolve withRejecter:(RCTPromiseRejectBlock)reject)
RCT_EXTERN_METHOD(setItem:(NSString *)key withValue:(NSString *)value withAppGroup:(NSString *)appGroup withResolver:(RCTPromiseResolveBlock)resolve withRejecter:(RCTPromiseRejectBlock)reject)

@end

File: ios/Widgetkit.swift
import WidgetKit

@objc(Widgetkit)
class Widgetkit: NSObject {
    
    @objc(reloadAllTimelines)
    func reloadAllTimelines() -> Void {
        if #available(iOS 14.0, *) {
            #if arch(arm64) || arch(i386) || arch(x86_64)
                WidgetCenter.shared.reloadAllTimelines()
            #endif
        }
    }

    @objc(reloadTimelines:)
    func reloadTimelines(ofKind: String) -> Void {
        if #available(iOS 14.0, *) {
            #if arch(arm64) || arch(i386) || arch(x86_64)
                WidgetCenter.shared.reloadTimelines(ofKind: ofKind)
            #endif
        }
    }

    @objc
    static func requiresMainQueueSetup() -> Bool {
        return true
    }
}
